src/component/handlers/composition/DraftEditorCompositionHandler.js

```diff
@@ -45,7 +45,17 @@ let textInputData = '';
 
 var DraftEditorCompositionHandler = {
   onBeforeInput: function(editor: DraftEditor, e: SyntheticInputEvent): void {
-    textInputData = (textInputData || '') + e.data;
+    // When we're consuming native events (detected by the lact of a `nativeEvent` accessor), we discard
```

Comment:
If you want to lock it down to just safari, [here's a good example of how to do that](https://github.com/textioHQ/draft-js/blob/release/src/component/utils/needsClipboardPolyfill.js#L14).

Reply:
It's more a question of whether we're consuming native events or not rather than the browser, however at the moment it's just Safari.